111-use-buildah-instead-of-kaniko.md

```diff
@@ -0,0 +1,236 @@
+
+As in case of Kaniko, Buildah will have also some allowed options that can be specified for the `build` and `push` phases:
+
+- `additionalBuildOptions`
+  - `--authfile` - path to the authentication file (usually created using the buildah login).
+  - `--cert-dir` - path to directory with certificates that should be used.
+  - `--creds` - the `[username[:password]]` to use to authenticate with the registry if required.
+  - `--decryption-key` - the `[key[:passphrase]]` to be used for decryption of images.
+  - `--retry` - number of times to retry in case of failure during image pull (the base one).
+  - `--retry-delay` - duration between retry attempts.
+  - `--tls-verify` - skip TLS verification for insecure registries.
+
+- `additionalPushOptions`
+  - `--authfile` - path to the authentication file (usually created using the `buildah login`).
+  - `--cert-dir` - path to directory with certificates that should be used for connecting to registries.
+  - `--creds` - the `[username[:password]]` to use to authenticate with the registry if required.
+  - `--retry` - number of times to retry in case of failure during image pull (the base one).
+  - `--retry-delay` - duration between retry attempts.
+  - `--tls-verify` - skip TLS verification for insecure registries.
+
+If the user-specified Buildah options will contain forbidden options (or not known), the `InvalidResourceException` will be thrown (logged inside the operator log and in the `.status` section of `KafkaConnect` CR) and the build will fail.
+
+Finally, this feature will be available on Kubernetes only - it will not be available on OpenShift, which is mentioned as one of the rejected alternative.
+So the implementation and usage of OpenShift Build API will be kept.
+
+### YAML example of configuring Connect Build with Buildah options
+
+The following YAML example shows how the `KafkaConnect` CR with build configuration (and Buildah additional options) will look like:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaConnect
+metadata:
+  name: my-connect-cluster
+  annotations:
+    strimzi.io/use-connector-resources: "true"
+spec:
+  version: 4.1.0
+  replicas: 1
+  bootstrapServers: my-cluster-kafka-bootstrap:9093
+  tls:
+    trustedCertificates:
+      - secretName: my-cluster-cluster-ca-cert
+        pattern: "*.crt"
+  config:
+    group.id: connect-cluster
+    offset.storage.topic: connect-cluster-offsets
+    config.storage.topic: connect-cluster-configs
+    status.storage.topic: connect-cluster-status
+    config.storage.replication.factor: -1
+    offset.storage.replication.factor: -1
+    status.storage.replication.factor: -1
+  build:
+    output:
+      type: docker
+      image: my-internal-registry:5000/repository/image:latest
+      additionalBuildOptions:
+        - "--tls-verify=false"
+      additionalPushOptions:
+        - "--quiet"
```

Comment:
I don't see option this in the list above?

Reply:
Good point, it seems that I accidentally removed it 🤦 . Thanks for pointing it out.